### DISCUSSION TITLE
overlayPreferenceValue Support

### DISCUSSION THREAD

**Discussion by gili-labs** [nalexn/ViewInspector#144] [Q&A] [answered]
Hi Alexei,

I read in the guide that overlayPreferenceValue and backgroundPreferenceValue block inspection, which confirms what I experienced yesterday when writing a test. The guide indicates this is under investigation. Have you made any progress with regards to this investigation?

Regards,
-Patrick

**Comment by nalexn** [comment]
Hey @gili-labs , I need to check if there are any changes internally for iOS 15, but I think the approach similar to inspecting an `alert` should work - that is, introduce a substitute function `overlayPreferenceValue2` that would capture the necessary context. I'll look into this for the next version. Not ideal, but definitely better than a complete block.

**Comment by nalexn** [comment, marked as answer]
Good news: the factual blocker was removed in one of the last SDK releases, so both `overlayPreferenceValue` and `backgroundPreferenceValue` will gain *full* support in the coming library update ([commit](https://github.com/nalexn/ViewInspector/commit/6a8e7959b2d8e59c16c019bc1301aaa3e917363b)). By full I mean also being able to get access to the content, including by the `find`, and all starting from SwiftUI v1 (iOS 13).

**Reply by nalexn** [reply]
Support has been added in v0.9.1
